src/Suave/Suave.fsproj

```diff
@@ -140,15 +179,7 @@
     <Reference Include="System.Web.Extensions" />
     <Reference Include="System.Xml" />
   </ItemGroup>
-  <Choose>
-    <When Condition="$(TargetFrameworkIdentifier) == '.NETFramework' And $(TargetFrameworkVersion) == 'v4.5'">
-      <ItemGroup>
-        <Reference Include="FSharp.Core">
-          <HintPath>..\..\packages\FSharp.Core\lib\net40\FSharp.Core.dll</HintPath>
-          <Private>True</Private>
-          <Paket>True</Paket>
-        </Reference>
-      </ItemGroup>
-    </When>
-  </Choose>
+  <Target Name="BeforeBuild" BeforeTargets="Build" Condition="'$(TargetPlatform)' == 'coreclr'">
+    <Exec Command="..\..\tools\nuget\NuGet.exe restore -Source https://www.myget.org/F/dotnet-core;https://www.myget.org/F/fsharp-daily;https://nuget.org/api/v2 project.json -PackagesDirectory ..\..\packages -Verbosity detailed" />

```

Comment:
Absolutely, but paket would need to download everything for all possible platforms, making the paket.dependecies more fragile, also the paths would need to be modified to remove the version info, or paket extended to add it.  Frankly, I would leave it until you have paket fixed, and just consider this a point in time solution.  My guess is paket will have the necessary functionality on Monday, you do have the weekend to implement it after all ☺

From: Steffen Forkmann [mailto:notifications@github.com]
Sent: Thursday, November 12, 2015 10:34 AM
To: SuaveIO/suave <suave@noreply.github.com>
Cc: Kevin Ransom <Kevin.Ransom@microsoft.com>
Subject: Re: [suave] Enable build and run on coreclr for suave (#334)


In src/Suave/Suave.fsproj<https://github.com/SuaveIO/suave/pull/334#discussion_r44695330>:

> @@ -140,15 +179,7 @@

>      <Reference Include="System.Web.Extensions" />

>      <Reference Include="System.Xml" />

>    </ItemGroup>

> -  <Choose>

> -    <When Condition="$(TargetFrameworkIdentifier) == '.NETFramework' And $(TargetFrameworkVersion) == 'v4.5'">

> -      <ItemGroup>

> -        <Reference Include="FSharp.Core">

> -          <HintPath>..\..\packages\FSharp.Core\lib\net40\FSharp.Core.dll</HintPath>

> -          <Private>True</Private>

> -          <Paket>True</Paket>

> -        </Reference>

> -      </ItemGroup>

> -    </When>

> -  </Choose>

> +  <Target Name="BeforeBuild" BeforeTargets="Build" Condition="'$(TargetPlatform)' == 'coreclr'">

> +    <Exec Command="..\..\tools\nuget\NuGet.exe restore -Source https://www.myget.org/F/dotnet-core;https://www.myget.org/F/fsharp-daily;https://nuget.org/api/v2<https://www.myget.org/F/dotnet-core;https:/www.myget.org/F/fsharp-daily;https:/nuget.org/api/v2> project.json -PackagesDirectory ..\..\packages -Verbosity detailed" />

Yes this helps a lot. Thanks.

Hypothetical question: if I commit the project.lock.json and the packages folder and remove project.lock, would the project still build?
I tried that with coreclr branch and it failed. There is still one little piece missing.

—
Reply to this email directly or view it on GitHub<https://github.com/SuaveIO/suave/pull/334/files#r44695330>.